File: solutions/csharp/sublist/2/Sublist.cs
using System;
using System.Collections.Generic;
using System.Linq;

// ReSharper disable once CheckNamespace
public static class Sublist
{
    public static SublistType Classify<T>(List<T> leftList, List<T> rightList)
        where T : IComparable
    {
        _ = leftList ?? throw new ArgumentNullException(nameof(leftList));
        _ = rightList ?? throw new ArgumentNullException(nameof(rightList));
        return leftList.SequenceEqual(rightList)
            ? SublistType.Equal
            : leftList.IsSuperlistOf(rightList)
                ? SublistType.Superlist
                : leftList.IsSublistOf(rightList)
                    ? SublistType.Sublist
                    : SublistType.Unequal;
    }
}

public static class ListExtensions
{
    public static bool IsSublistOf<T>(this IReadOnlyCollection<T> leftList, IReadOnlyCollection<T> rightList)
        where T : IComparable
        => rightList.IsSuperlistOf(leftList);

    public static bool IsSuperlistOf<T>(this IReadOnlyCollection<T> leftList, IReadOnlyCollection<T> rightList)
        where T : IComparable
        => leftList.Count >= rightList.Count
           && (rightList.Count == 0 || IsSublistContainedIn(leftList, rightList));

    private static bool IsSublistContainedIn<T>(IReadOnlyCollection<T> leftList, IReadOnlyCollection<T> rightList)
        where T : IComparable
        => Enumerable.Range(0, leftList.Count - rightList.Count + 1)
            .Any(index => leftList.IsEqualAt(index, rightList));

    private static bool IsEqualAt<T>(this IEnumerable<T> leftList, int index, IReadOnlyCollection<T> rightList)
        where T : IComparable
        => leftList.Skip(index)
            .Take(rightList.Count)
            .SequenceEqual(rightList);
}

public enum SublistType
{
    Equal,
    Unequal,
    Superlist,
    Sublist
}

File: solutions/python/sublist/2/sublist.py
SUBLIST = "sublist"
SUPERLIST = "superlist"
EQUAL = "equal"
UNEQUAL = "unequal"

def is_sublist(smaller_list, larger_list):
    if not smaller_list:
        return True

    if len(smaller_list) > len(larger_list):
        return False

    return _contains_sublist(smaller_list, larger_list)

def _contains_sublist(smaller_list, larger_list):
    last_start_index = len(larger_list) - len(smaller_list)
    return any(
        _contains_at_index(smaller_list, larger_list, start_index)
        for start_index in range(last_start_index + 1)
    )

def _contains_at_index(smaller_list, larger_list, start_index):
    end_index = start_index + len(smaller_list)
    return larger_list[start_index:end_index] == smaller_list

def sublist(list_one, list_two):
    if list_one == list_two:
        return EQUAL

    if is_sublist(list_one, list_two):
        return SUBLIST

    if is_sublist(list_two, list_one):
        return SUPERLIST

    return UNEQUAL